File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/response/type/Tenant.kt
package gaia.sdk.response.type

import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.response.intf.*
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator
import gaia.sdk.request.enumeration.*

/**
* Represents tenant information
*/
data class Tenant @JsonCreator constructor(
    /**
    * The tenant id
    */
    @JsonProperty("tenantId") val tenantId:Uuid? = null, 
    /**
    * The name of the tenant
    */
    @JsonProperty("qualifier") val qualifier:String? = null
)

File: gaia-sdk-python/gaia_sdk/graphql/response/type/Tenant.py


from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class Tenant:
    """
    Represents tenant information
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    """
    The tenant id
    """
    @property
    def tenant_id(self) -> Uuid:
        return Uuid(self.dictionary.get("tenantId"))
    """
    The name of the tenant
    """
    @property
    def qualifier(self) -> String:
        return String(self.dictionary.get("qualifier"))
    """
    The list of implicit identities
    """
    @property
    def implicit_identities(self) -> List[String]:
        return list(map(lambda x: String(x), self.dictionary.get("implicitIdentities")))
    """
    The list of explicit identities
    """
    @property
    def explicit_identities(self) -> List[String]:
        return list(map(lambda x: String(x), self.dictionary.get("explicitIdentities")))
